addons/hr/tests/test_hr_employee.py

```diff
@@ -250,6 +253,34 @@ def test_availability_user_infos_employee(self):
         for field in user_fields:
             self.assertEqual(employee[field], user[field])
 
+    def test_set_user_on_new_employee(self):
+        test_company = self.env['res.company'].create({
+            'name' : 'Test User Company',
+        })
+        self.env['hr.employee'].create({
+            'name': 'Hr Officer - employee',
+            'user_id': self.res_users_hr_officer.id,
+            'company_id': test_company.id,
+        })
+
+        self.res_users_hr_officer.write({'company_ids': test_company.ids, 'company_id': test_company.id})
+        employee_form = Form(self.env['hr.employee'].with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```

Comment:
Maybe write comment like:
```suggestion
        # Try to set the user with existing employee in the company, on a new employee form
        employee_form = Form(self.env['hr.employee'].with_user(self.res_users_hr_officer).with_company(company=test_company.id))
```